x-pack/plugins/enterprise_search/public/applications/app_search/components/search_ui/search_ui_logic.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+interface InitialFieldValues {
+  validFields: string[];
+  validSortFields: string[];
+  validFacetFields: string[];
+}
+interface SearchUIActions {
+  initializeData(): void;
+  dataInitialized(initialFieldValues: InitialFieldValues): InitialFieldValues;
+  titleFieldChanged(titleField: string): { titleField: string };
+  URLFieldChanged(urlField: string): { urlField: string };
+  facetFieldsChanged(facetFields: string[]): { facetFields: string[] };
+  sortFieldsChanged(sortFields: string[]): { sortFields: string[] };
+  activeFieldChanged(activeField: string): { activeField: string };
+}
+
+interface SearchUIValues {
+  dataLoading: boolean;
+  validFields: string[];
+  validSortFields: string[];
+  validFacetFields: string[];
+  titleField: string;
+  urlField: string;
+  facetFields: string[];
+  sortFields: string[];
+  activeField: string;
+}
+
+export const SearchUILogic = kea<MakeLogicType<SearchUIValues, SearchUIActions>>({
+  path: ['enterprise_search', 'app_search', 'search_ui_logic'],
+  actions: () => ({
+    initializeData: () => true,
+    dataInitialized: (initialFieldValues) => initialFieldValues,
+    titleFieldChanged: (titleField) => ({ titleField }),
+    URLFieldChanged: (urlField) => ({ urlField }),
+    facetFieldsChanged: (facetFields) => ({ facetFields }),
+    sortFieldsChanged: (sortFields) => ({ sortFields }),
+    activeFieldChanged: (activeField) => ({ activeField }),
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        dataInitialized: () => false,
+      },
+    ],
+    validFields: [[], { dataInitialized: (_, { validFields }) => validFields }],
+    validSortFields: [[], { dataInitialized: (_, { validSortFields }) => validSortFields }],
+    validFacetFields: [[], { dataInitialized: (_, { validFacetFields }) => validFacetFields }],
```

Comment:
[naming question] Just to check, is there a specific meaning behind the `valid` prefix var name here? i.e.: is there a concept of _invalid_ fields that we're storing in state?

If there isn't, maybe consider renaming this to just `fields`/`sortFields`/`facetFields`? 🤷

Reply:
`validSortFields` are all of the fields that could be selected in the sort dropdown, `sortFields` are the actually selected fields. This is just carry over from the old UI so I don't feel strongly about it. Other options:

`allSortFields` and `sortFields`
`sortFields` and `selectedSortFields`

Reply:
Ohh oops, I missed the actual `sortFields`/`facetFields` values 🤦‍♀️  I like `allSortFields` and `sortFields`!

Reply:
Ah, so, this data comes back from the server with the `valid` prefix. For that reason, I'll keep it as is.

FWIW, I think they're prefixed with `valid` because there's only certain fields that are "valid" to use for each of these options, otherwise we could just use a single list of fields from the schema to populate each dropdown.

Reply:
Gotcha, no worries!